components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
-    zero = 0._r8
-    nstep = get_nstep()
-    call check_tracers_init(state, tracerint)
+  ! Adjust the surface fluxes to reduce instabilities in near sfc layer
+  if (phys_do_flux_avg()) call flux_avg_run(state, cam_in,  pbuf, nstep, ztodt)
 
-!!== KZ_WCON
+  ! Validate the physics state.
+  if (state_debug_checks) call physics_state_check(state, name="before tphysac")
 
-    call check_qflx(state, tend, "PHYAC01", nstep, ztodt, cam_in%cflx(:,1))
+  ! Associate pointers with physics buffer fields
+  itim_old = pbuf_old_tim_idx()
 
-#ifndef MMF_FLUX_BYPASS
-    if(.not.use_qqflx_fixer) then 
+  call pbuf_get_field(pbuf, pbuf_get_index('DTCORE'), dtcore, start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
 
-       ! Check if latent heat flux exceeds the total moisture content of the
-       ! lowest model layer, thereby creating negative moisture.
-#ifdef MAML
-       call qneg4('TPHYSAC '       ,lchnk               ,ncol  ,ztodt ,               &
-            num_inst_atm,state%q(1,pver,1),state%rpdel(1,pver) ,cam_in%shf(:,:) ,         &
-            cam_in%lhf(:,:) , cam_in%cflx )
-#else
-       call qneg4('TPHYSAC '       ,lchnk               ,ncol  ,ztodt ,               &
-            state%q(1,pver,1),state%rpdel(1,pver) ,cam_in%shf ,         &
-            cam_in%lhf , cam_in%cflx )
-#endif 
+  call pbuf_get_field(pbuf, tini_idx, tini)
+  call pbuf_get_field(pbuf, qini_idx, qini)
+  call pbuf_get_field(pbuf, cldliqini_idx, cldliqini)
+  call pbuf_get_field(pbuf, cldiceini_idx, cldiceini)
 
-    end if 
-#endif
+  call pbuf_get_field(pbuf, pbuf_get_index('CLD'),cld,start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
+  call pbuf_get_field(pbuf, pbuf_get_index('AST'),ast,start=(/1,1,itim_old/),kount=(/pcols,pver,1/))
 
-    call check_qflx(state, tend, "PHYAC02", nstep, ztodt, cam_in%cflx(:,1))
+  !-----------------------------------------------------------------------------
+  ! emissions of aerosols and gas-phase chemistry constituents at surface
+  !-----------------------------------------------------------------------------
+  if (l_tracer_aero) then
+    call chem_emissions( state, cam_in )
+  end if
 
-!!== KZ_WCON
+  !-----------------------------------------------------------------------------
+  ! get nstep and zero array for energy checker
+  !-----------------------------------------------------------------------------
+  zero = 0._r8
+  nstep = get_nstep()
+  call check_tracers_init(state, tracerint)
 
-    call t_stopf('tphysac_init')
+  ! Check if LHF exceeds the total moisture content of the lowest layer
+  call qneg4('TPHYSAC ', lchnk, ncol, ztodt, &
+              state%q(1,pver,1), state%rpdel(1,pver), &
+              cam_in%shf, cam_in%lhf, cam_in%cflx )
 
-if (l_tracer_aero) then
-    !===================================================
-    ! Source/sink terms for advected tracers.
-    !===================================================
-    call t_startf('adv_tracer_src_snk')
```

Comment:
We lost some more timers here. I don't think this is a good thing.

Reply:
added back